src/hotspot/share/runtime/frame.cpp

```diff
@@ -1162,10 +1162,8 @@ void frame::oops_do_internal(OopClosure* f, NMethodClosure* cf,
                              const RegisterMap* map, bool use_interpreter_oop_map_cache) const {
 #ifndef PRODUCT
   // simulate GC crash here to dump java thread in error report
-  if (CrashGCForDumpingJavaThread) {
-    char *t = nullptr;
-    *t = 'c';
-  }
+  if (CrashGCForDumpingJavaThread)
+    guarantee(!CrashGCForDumpingJavaThread, "");
```

Comment:
Besides being contrary to the HotSpot Style Guide, that isn't at all what I
meant.  We don't need to test CrashGCForDumpingJavaThread before testing it
again int the guarantee.

Reply:
Sorry, got it now.
The duplicate `CrashGCForDumpingJavaThread` check has been removed.

Reply:
1. --with-debug-level=optimized: `java -XX:+CrashGCForDumpingJavaThread -version` do not trigger jvm crash and print version meaasge.
2. --with-debug-level=release: `java -XX:+CrashGCForDumpingJavaThread -version` report VM option 'CrashGCForDumpingJavaThread' is develop and is available only in debug version of VM.
3. --with-debug-level=fastdebug: `java -XX:+CrashGCForDumpingJavaThread -version` trigger jvm crash correctly.

Reply:
Good.  Those results are as expected.  In case it isn't clear, the reason for the difference between optimized
(no crash) and fastdebug (crash) is that optimized defaults VerifyBeforeExit to false, while for fastdebug it
defaults to true.  Enabling that option in an optimized build will crash in the same manner as fastdebug.

Reply:
Thanks for the detail explanation.